tests/datavolume_test.go

```diff
@@ -884,6 +884,88 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
+			Expect(err).ToNot(HaveOccurred())
+
+			By("verifying pvc was created")
+			pvc, err := utils.WaitForPVC(f.K8sClient, dataVolume.Namespace, dataVolume.Name)
+			Expect(err).ToNot(HaveOccurred())
+			f.ForceBindIfWaitForFirstConsumer(pvc)
+
+			By("find importer pod")
+			pod, err := utils.FindPodByPrefix(f.K8sClient, dataVolume.Namespace, common.ImporterPodName, common.CDILabelSelector)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(pod).ToNot(BeNil())
+			By("verifying passed annotation")
+			Expect(pod.Annotations[controller.AnnPodNetwork]).To(Equal("net1"))
+			By("verifying non-passed annotation")
+			Expect(pod.Annotations["annot1"]).ToNot(Equal("value1"))
+		})
+		It("[test_id:5365]Uploader pod should have specific datavolume annotations passed but not others", func() {
+			dataVolume := utils.NewDataVolumeForUpload(dataVolumeName, "1Gi")
+			By(fmt.Sprintf("creating new datavolume %s with annotations", dataVolume.Name))
+			dataVolume.Annotations[controller.AnnPodNetwork] = "net1"
+			dataVolume.Annotations["annot1"] = "value1"
+			dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("verifying pvc was created")
+			pvc, err := utils.WaitForPVC(f.K8sClient, dataVolume.Namespace, dataVolume.Name)
+			Expect(err).ToNot(HaveOccurred())
+			f.ForceBindIfWaitForFirstConsumer(pvc)
+
+			By("find uploader pod")
+			pod, err := utils.FindPodByPrefix(f.K8sClient, dataVolume.Namespace, "cdi-upload", common.CDILabelSelector)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(pod).ToNot(BeNil())
+			By("verifying passed annotation")
+			Expect(pod.Annotations[controller.AnnPodNetwork]).To(Equal("net1"))
+			By("verifying non-passed annotation")
+			Expect(pod.Annotations["annot1"]).ToNot(Equal("value1"))
+		})
+		It("[test_id:5366]Cloner pod should have specific datavolume annotations passed but not others", func() {
+			sourcePodFillerName := fmt.Sprintf("%s-filler-pod", dataVolumeName)
+			pvcDef := utils.NewPVCDefinition(pvcName, "1Gi", nil, nil)
+			sourcePvc = f.CreateAndPopulateSourcePVC(pvcDef, sourcePodFillerName, fillCommand)
+			dataVolume := utils.NewCloningDataVolume(dataVolumeName, "1Gi", sourcePvc)
+
+			By(fmt.Sprintf("creating new datavolume %s with annotations", dataVolume.Name))
+			dataVolume.Annotations[controller.AnnPodNetwork] = "net1"
+			dataVolume.Annotations["annot1"] = "value1"
+			dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("verifying pvc was created")
+			pvc, err := utils.WaitForPVC(f.K8sClient, dataVolume.Namespace, dataVolume.Name)
+			Expect(err).ToNot(HaveOccurred())
+			f.ForceBindIfWaitForFirstConsumer(pvc)
+
+			By("find source pod")
+			pod, err := utils.FindPodBysuffix(f.K8sClient, dataVolume.Namespace, "source-pod", common.CDILabelSelector)
```

Comment:
So I am concerned this pod could be gone before we get to finding it. The amount of data we are cloning is very small, and thus the pod could disappear before we get a chance to look at it. This would obviously not always happen, but the potential is certainly there. Maybe for this particular test generate more data so a clone will take longer?

Reply:
done